src/proba_orientation.cpp

```diff
@@ -251,39 +291,39 @@ vector<ProbaArray> getOriProbasList(const vector<Triple>& triples,
       bool remove_triple = false;
       if (X != -1) {
         if (triples[i][0] == X && triples[i][1] == Z) {
-          updateProba(1, 0, x2z, z2x, latent, propagation, probas_list[i],
+          updateProba(1, 0, x2z, z2x, latent_orientation, propagation, probas_list[i],
               probas_list2[i], need_propagation, remove_triple);
         } else if (triples[i][0] == Z && triples[i][1] == X) {
-          updateProba(0, 1, x2z, z2x, latent, propagation, probas_list[i],
+          updateProba(0, 1, x2z, z2x, latent_orientation, propagation, probas_list[i],
               probas_list2[i], need_propagation, remove_triple);
         } else if (triples[i][2] == X && triples[i][1] == Z) {
-          updateProba(2, 3, x2z, z2x, latent, propagation, probas_list[i],
+          updateProba(2, 3, x2z, z2x, latent_orientation, propagation, probas_list[i],
               probas_list2[i], need_propagation, remove_triple);
         } else if (triples[i][2] == Z && triples[i][1] == X) {
-          updateProba(3, 2, x2z, z2x, latent, propagation, probas_list[i],
+          updateProba(3, 2, x2z, z2x, latent_orientation, propagation, probas_list[i],
               probas_list2[i], need_propagation, remove_triple);
         }
         if (need_propagation)
-          propagate(latent, propagation, I3_list[i], probas_list[i],
+          propagate(latent_orientation, propagation, I3_list[i], probas_list[i],
               probas_list2[i], score[i], log_score[i]);
       }    // if (X != -1)
       need_propagation = false;
       if (Y != -1) {
         if (triples[i][0] == Y && triples[i][1] == Z) {
-          updateProba(1, 0, y2z, z2y, latent, propagation, probas_list[i],
+          updateProba(1, 0, y2z, z2y, latent_orientation, propagation, probas_list[i],
               probas_list2[i], need_propagation, remove_triple);
         } else if (triples[i][0] == Z && triples[i][1] == Y) {
-          updateProba(0, 1, y2z, z2y, latent, propagation, probas_list[i],
+          updateProba(0, 1, y2z, z2y, latent_orientation, propagation, probas_list[i],
               probas_list2[i], need_propagation, remove_triple);
         } else if (triples[i][2] == Y && triples[i][1] == Z) {
-          updateProba(2, 3, y2z, z2y, latent, propagation, probas_list[i],
+          updateProba(2, 3, y2z, z2y, latent_orientation, propagation, probas_list[i],
               probas_list2[i], need_propagation, remove_triple);
         } else if (triples[i][2] == Z && triples[i][1] == Y) {
-          updateProba(3, 2, y2z, z2y, latent, propagation, probas_list[i],
+          updateProba(3, 2, y2z, z2y, latent_orientation, propagation, probas_list[i],
               probas_list2[i], need_propagation, remove_triple);
         }
         if (need_propagation)
-          propagate(latent, propagation, I3_list[i], probas_list[i],
+          propagate(latent_orientation, propagation, I3_list[i], probas_list[i],
```

Comment:
Not sure why you include the variable `environment.latent` here, the only relevant variable should be `latent_orientation` which is set correctly at the call site, so inside this function `latent` is unambiguous. So i suggest remove those changes.